### PR TITLE
Support building on Mono on Unix.

### PR DESCRIPTION
**Pull request by ddunkin**
Adds a bash bootstrapper for Cake.
Modifies build.cake to work with Cake on Mono.
Tested on macOS.

### CODE REVIEW

**Comment by bgrainger** [comment]
Do you know what it would take to update travis.yml to use build.sh? Would also have to install mono on the Travis CI image?

**Comment by ejball** [comment]
No string interpolation? :-(

File: build.cake [hunk 1]
```diff
@@ -92,21 +93,27 @@ Task("NuGetPublish")
...
+	{
+		arguments = exePath + " " + arguments;
+		exePath = "mono";
+	}
+	int exitCode = StartProcess(exePath, arguments);
 	if (exitCode == 1 && verify)
 		throw new InvalidOperationException("Generated docs don't match; use -target=GenerateDocs to regenerate.");
 	else if (exitCode != 0)
-		throw new InvalidOperationException($"Docs generation failed with exit code {exitCode}.");
+		throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Docs generation failed with exit code {0}.", exitCode));
```
Comment (ejball) [inline build.cake]: I know, it's seriously ingrained, but I think `InvariantCulture` is overkill here, especially for an exception message.

File: build.cake [hunk 2]
```diff
@@ -9,7 +10,7 @@ var trigger = Argument("trigger", "");
 
 var nugetSource = "https://api.nuget.org/v3/index.json";
 var solutionFileName = "Faithlife.Utility.sln";
-var docsAssembly = $@"src\Faithlife.Utility\bin\{configuration}\net46\Faithlife.Utility.dll";
+var docsAssembly = File("src/Faithlife.Utility/bin/" + configuration + "/net46/Faithlife.Utility.dll").ToString();
```
Comment (ejball) [inline build.cake]: Would `string.Format` be easier to read, here and below? I guess I'm not sure...

File: build.cake [hunk 3]
```diff
@@ -35,11 +36,11 @@ Task("Rebuild")
 
 Task("GenerateDocs")
 	.IsDependentOn("Build")
-	.Does(() => GenerateDocs(verify: false));
+	.Does(() => GenerateDocs(docsAssembly, docsSourceUri, verify: false));
 
 Task("VerifyGenerateDocs")
 	.IsDependentOn("Build")
-	.Does(() => GenerateDocs(verify: true));
+	.Does(() => GenerateDocs(docsAssembly, docsSourceUri, verify: true));
```
Comment (ejball) [inline build.cake]: Why was this necessary? Weird Mono issue?

Comment (ddunkin) [reply build.cake]: Script-scoped variables can't be referenced from functions when running with the Mono script host.

File: .travis.yml [hunk 4]
```diff
@@ -9,4 +10,4 @@ before_install:
 - sudo apt-get install dotnet-sdk-2.0.0
 
 script:
-- dotnet test tests/Faithlife.Utility.Tests -c Release /p:SourceLinkCreate=false /p:SourceLinkTest=false
+- ./build.sh --target=NuGetPackage
```
Comment (bgrainger) [inline .travis.yml]: I don't know why SourceLink is failing on Travis (seems to work OK in octokit, for example), but I think this MSBuild property hack could be replicated by changing the `"Build"` task as follows:

```
		var buildSettings = new DotNetCoreBuildSettings { Configuration = configuration, ArgumentCustomization = args => args.Append("--verbosity normal") };
		if (Context.Environment.Platform.IsUnix())
		{
			buildSettings.MSBuildSettings = new DotNetCoreMSBuildSettings
			{
				Properties =
				{
					{ "SourceLinkCreate", new[] { "false" } },
					{ "SourceLinkTest", new[] { "false" } },
				}
			};
		}
		DotNetCoreBuild(solutionFileName, buildSettings);
```

That would probably turn it off for macOS too; don't know if that's a concern. (Not sure if/how Cake distinguishes between Linux and macOS.)

Comment (bgrainger) [reply .travis.yml]: That's unlikely to fix the `net46`-related errors in the current PR build though.

Comment (bgrainger) [reply .travis.yml]: @ejball suggests that sourcelink fails because it [requires netcoreapp1.0](https://github.com/ctaggart/SourceLink/blob/master/dotnet-sourcelink-git/dotnet-sourcelink-git.csproj#L5) and that SDK isn't installed.

We could modify the travis.yml file to install it, or use the "hack" above. Thoughts?

Comment (bgrainger) [reply .travis.yml]: Maybe follow these steps: https://codeblog.jonskeet.uk/2017/08/15/using-net-core-2-0-sdk-on-travis/

Comment (ddunkin) [reply .travis.yml]: I figured it out on my own, then stumbled across Jon Skeet's post, then your comments here. 😩

**Comment by bgrainger** [comment]
I just pushed XmlDocMarkdown 0.5.4 to `master` and updated a couple of generated pages.

**Comment by ddunkin** [comment]
The Travis build passed. Without objection, I'll squash-merge this, @bgrainger.
